### PR TITLE
Test utils / CI fix

### PR DESCRIPTION
**Pull request by joshuamorton**
This PR:
- runs pytests in boxed threads, which ensures that separate tests won't interfere with each other through global `roscore` processes
- fixes issue where timeout errors were being swallowed by the generator in `wait_for_message` never evaluating the yielded object.
- fixes issue where `functools.partial` was being used for a callback that only had one arg
- creates boiler plate test for `controller` node

Other minor things this fixes:
- use `raw_input` on inputer node. We had moved to `input` when we thought py3 was a possibility.
- ensures that we're sourcing the correct files in `ci_scripts/*` by not using relative paths

### CODE REVIEW

File: buzzmobile/tests/unit/plan/controller_pytest_output.txt [hunk 1]
```diff
@@ -0,0 +1,84 @@
+============================= test session starts ==============================
```
Comment (joshuamorton) [inline buzzmobile/tests/unit/plan/controller_pytest_output.txt]: delete this file.

File: buzzmobile/tests/unit/plan/test_controller.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    def test_state(self):
+         with mock_pub('/joy', Joy, queue_size=None) as joy_node:
+            
+            # TODO When this line is enabled, it causes the bearing test to
+            # fail stating the message has not been sent yet.
+            # Even though there is an await in the bearing test.
+            # Also, it doesn't matter which topic is checked
+            with check_topic('/buzzmobile/car_state', CarState) as cs:
+                # Ensure starts in Start Mode
+                joy_node.send(Joy())
```
Comment (joshuamorton) [inline buzzmobile/tests/unit/plan/test_controller.py]: may not need this line?

File: buzzmobile/tests/unit/process/test_bearing.py [hunk 3]
```diff
@@ -29,6 +29,9 @@ def test_bearing_node(self):
                 fix_node.send(NavSatFix(None, None, 39.029128, -111.838257, None, None, None))
 
             # check the output from the node
-            await(ct.wait_for_message())
-            assert np.isclose(ct.message.data, 1.19212)
+                await(ct.wait_for_message())
+                print(ct._message)
+                assert ct._message
+            #assert np.isclose(ct.message.data, 1.19212)
```
Comment (joshuamorton) [inline buzzmobile/tests/unit/process/test_bearing.py]: Fix this.

File: buzzmobile/tests/test_utils/mock_nodes.py [hunk 4]
```diff
@@ -102,16 +101,20 @@ def cb_wrapper(message):
...
-    rospy.core._in_shutdown = False  # pylint: disable=protected-access
+        return callback(message)
+
+    sub = rospy.Subscriber(topic, rosmsg_type, cb_wrapper)
+    try:
+        yield test_node
+    finally:
+        sub.unregister()
+        rospy.signal_shutdown('test complete')
+        reload(rospy)
```
Comment (joshuamorton) [inline buzzmobile/tests/test_utils/mock_nodes.py]: I don't think this is needed.

File: buzzmobile/tests/test_utils/mock_nodes.py [hunk 5]
```diff
@@ -102,16 +101,20 @@ def cb_wrapper(message):
...
-    rospy.client._init_node_args = None  # pylint: disable=protected-access
-    rospy.core._shutdown_flag = False  # pylint: disable=protected-access
-    rospy.core._in_shutdown = False  # pylint: disable=protected-access
+        return callback(message)
+
+    sub = rospy.Subscriber(topic, rosmsg_type, cb_wrapper)
+    try:
+        yield test_node
+    finally:
+        sub.unregister()
```
Comment (joshuamorton) [inline buzzmobile/tests/test_utils/mock_nodes.py]: I don't think this is needed.

File: buzzmobile/tests/test_utils/mock_nodes.py [hunk 6]
```diff
@@ -102,16 +101,20 @@ def cb_wrapper(message):
...
+        rospy.signal_shutdown('test complete')
+        reload(rospy)
+
+        # Ros really doesn't want you to reinitialize a node once it's been
+        # shutdown because there can be bad side effects, but we are good
+        # at cleaning up after ourselves.
+        rospy.client._init_node_args = None  # pylint: disable=protected-access
+        rospy.core._shutdown_flag = False  # pylint: disable=protected-access
+        rospy.core._in_shutdown = False  # pylint: disable=protected-access
```
Comment (joshuamorton) [inline buzzmobile/tests/test_utils/mock_nodes.py:116]: Interestingly, with the xdist based testing, none of this cleanup may be needed.

File: buzzmobile/tests/test_utils/mock_nodes.py (hunk 4 above)
Comment (irapha) [reply buzzmobile/tests/test_utils/mock_nodes.py]: done

File: buzzmobile/tests/test_utils/mock_nodes.py (hunk 6 above)
Comment (irapha) [reply buzzmobile/tests/test_utils/mock_nodes.py:116]: ack

File: buzzmobile/tests/unit/plan/controller_pytest_output.txt (hunk 1 above)
Comment (irapha) [reply buzzmobile/tests/unit/plan/controller_pytest_output.txt]: done

File: buzzmobile/tests/test_utils/rostest_utils.py [hunk 7]
```diff
@@ -43,14 +43,13 @@ def await(gen):
...
+    it will wait for the `wait_for_message` call to finish running.
 
     In other words, if await(Node) doesn't raise an error, you should have
     successfully recieved a message, and can therefore test against it,
     otherwise the message access can raise a NoMessage error.
     """
+    res = None
     for item in gen:
-        yield item
+        res = item
```
Comment (iRapha) [inline buzzmobile/tests/test_utils/rostest_utils.py]: Wait why is this needed?

Comment (joshuamorton) [inline buzzmobile/tests/test_utils/rostest_utils.py:54]: I think it could just as well be

```
for _ in gen:
    _
```
but I wanted to be able to return the result. The yield change is important. I was being dumb before.

**Review by irapha — APPROVED** [review]
